src/Microsoft.NetCore.Analyzers/Core/Security/DoNotUseInsecureDeserializerJsonNetWithoutBinder.cs

```diff
@@ -0,0 +1,242 @@
+            });
+
+        private static readonly PropertyMapperCollection PropertyMappers = new PropertyMapperCollection(
+            new PropertyMapper(
+                "TypeNameHandling",
+                (ValueContentAbstractValue valueContentAbstractValue) =>
+                    PropertySetCallbacks.EvaluateLiteralValues(
+                        valueContentAbstractValue,
+                        (object o) => o is int i && i != 0),   // None is 0, and anything other than None is flagged.
+                TypeNameHandlingIndex),
+            new PropertyMapper(
+                "Binder",
+                PropertySetCallbacks.FlagIfNull,
+                SerializationBinderIndex),      // Binder & SerializationBinder have the same underlying value.
+            new PropertyMapper(
+                "SerializationBinder",
+                PropertySetCallbacks.FlagIfNull,
+                SerializationBinderIndex));     // Binder & SerializationBinder have the same underlying value.
+
+        private static readonly HazardousUsageEvaluatorCollection HazardousUsageEvaluators = new HazardousUsageEvaluatorCollection(
+            SecurityHelpers.JsonSerializerDeserializationMethods.Select(
+                (string methodName) => new HazardousUsageEvaluator(
+                    methodName,
+                    PropertySetCallbacks.HazardousIfAllFlaggedAndAtLeastOneKnown))
+                .Concat(
+                    new HazardousUsageEvaluator(
+                        HazardousUsageEvaluatorKind.Initialization,
+                        PropertySetCallbacks.HazardousIfAllFlaggedAndAtLeastOneKnown)));
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(
+                (CompilationStartAnalysisContext compilationStartAnalysisContext) =>
+                {
+                    WellKnownTypeProvider wellKnownTypeProvider = WellKnownTypeProvider.GetOrCreate(compilationStartAnalysisContext.Compilation);
+                    if (!wellKnownTypeProvider.TryGetTypeByMetadataName(
+                            WellKnownTypeNames.NewtonsoftJsonJsonSerializer,
+                            out INamedTypeSymbol jsonSerializerSymbol))
+                    {
+                        return;
+                    }
+
+                    PooledHashSet<(IOperation Operation, ISymbol ContainingSymbol)> rootOperationsNeedingAnalysis = PooledHashSet<(IOperation, ISymbol)>.GetInstance();
+
+                    compilationStartAnalysisContext.RegisterOperationBlockStartAction(
+                        (OperationBlockStartAnalysisContext operationBlockStartAnalysisContext) =>
+                        {
```

Comment:
Once https://github.com/dotnet/roslyn-analyzers/pull/2699 is merged, you need to follow the pattern used in there for DFA rules to honor `IsConfiguredToSkipAnalysis` for context's OwningSymbol.

It would be good if `PropertySetAnalysis.BatchGetOrComputeHazardousUsages` can be enhanced to add an assert similar to https://github.com/dotnet/roslyn-analyzers/pull/2699/files#diff-7f3f64f1f55afeb72c432e12a305385aR53 for future DFA rules using this analysis:
```csharp
Debug.Assert(!owningSymbol.IsConfiguredToSkipAnalysis(analyzerOptions, rule, wellKnownTypeProvider.Compilation, cancellationToken));
```